File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int t0 = 0, t1 = 1, t2 = 1, t3;
        switch(n) {
            case 0:
                return t0;
            case 1:
                return t1;
            case 2:
                return t2;
        }
        for(int i = 3; i <= n; i++) {
            t3 = t2 + t1 + t0;
            t0 = t1;
            t1 = t2;
            t2 = t3;
        }
        return t3;
    }
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        int zero = 0, one = 1, two = 1, next = 2;
        if(n == 0)
            return 0;
        if(n == 1 || n == 2)
            return 1;
        for(int i = 0; i < n - 2; ++i) {
            next = zero + one + two;
            zero = one;
            one = two;
            two = next;
        }
        return next;
    }
}